simplesat/tests/adhoc_allow_newer_older_any.yaml

```diff
@@ -0,0 +1,55 @@
+packages:
+  - A 0.0.1-1;
+  - A 1.0.0-1;
+  - A 2.0.0-1;
+  - A 2.1.0-1;
+  - B 1.0.0-1; depends (A ^= 1.0, A != 2.1.0-1)
+  - B 2.0.0-1; depends (A != 2.1.0-1)
+  - C 1.0.0-1; depends (B > 2.0.0-1)
+  - D 1.0.0-1;
+  - E 1.0.0-1; depends (C > 2.0, D < 1.0)
+
+installed:
+  - A 0.0.1-1
+
+adhoc:
+  allow_newer:
+    - A
+  allow_older:
+    - B
+  allow_any:
+    - C
+    - D
+
+request:
+  - operation: "install"
+    requirement: "B ^= 1.0.0"
+  - operation: "install"
+    requirement: "E"
+
+transaction:
+  - kind: "remove"
+    package: "A 0.0.1-1"
+  - kind: "install"
+    package: "A 2.0.0-1"
```

Comment:
isn't that an implementation detail ? Strictly speaking, one could get another solution which includes `A ^= 1.0.0`, right ? Does it make sense to add in `request` an `install A ^= 2.0.0` operation ?